tests/suites/test_suite_psa_crypto.function

```diff
@@ -4336,6 +4501,23 @@ void aead_multipart_state_test( int key_type_arg, data_t *key_data,
 
     PSA_ASSERT( psa_aead_set_nonce( &operation, nonce->x, nonce->len ) );
 
+    PSA_ASSERT( psa_aead_update_ad( &operation, additional_data->x,
+                                    additional_data->len ) );
+
+    TEST_EQUAL( psa_aead_set_lengths( &operation, additional_data->len,
+                                      input_data->len ),
+                PSA_ERROR_BAD_STATE );
+
+    psa_aead_abort( &operation );
+
+    /* ------------------------------------------------------- */
+
+    operation = psa_aead_operation_init( );
```

Comment:
That's not outright wrong, but it's superfluous and arguably inferior: `psa_aead_abort` is supposed to leave the object in a state that's equivalent to a freshly-initialized object. Without the reassignments to `operation`, this function would be calling `setup` after `abort`, which is a good way to test `abort`.